File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int min = Integer.MAX_VALUE;
        int max = 0;

        for(int i = 0; i<prices.length; i++){
            min = Math.min(min, prices[i]);
            int profit = prices[i] - min;
            max = Math.max(max, profit);
        }

        return max;
    }
}

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int minP = INT_MAX;
        int maxP = INT_MIN;

        for(int i = 0; i<prices.size(); i++){
            minP = min(minP, prices[i]);

            int profit = prices[i] - minP;

            maxP = max(maxP, profit);
        }

        return maxP;
    }
};